PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/model/PhylogeneticTreeNode.java

```diff
@@ -1,109 +1,240 @@
 package nl.tudelft.pl2016gr2.gui.model;
 
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode;
+
 import java.util.ArrayList;
-import java.util.Objects;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
 
 /**
- * This class is an adapter class which maps the methods of the
- * net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode class to the interface which is needed by
- * our application.
+ * This class is a storage class for a phylogenetic tree node. It retrieves and stores the needed
+ * values of the newick parser {@link TreeNode}.
  *
  * @author Faris
  */
-public class PhylogeneticTreeNode implements IPhylogeneticTreeNode {
+public class PhylogeneticTreeNode implements IPhylogeneticTreeNode, Iterable<PhylogeneticTreeNode> {
```

Comment:
It is only used by the root node to parse over all of the leaf nodes and add them to a (genomename -> node) mapping. It is not used by any other classes/for any other purpose.
I used an iterator because it is well known (so people know what it does). The only thing which might be a little bit different from the usual iterator is that it only parses over the leaf nodes (so non-leaf nodes of the tree are skipped).